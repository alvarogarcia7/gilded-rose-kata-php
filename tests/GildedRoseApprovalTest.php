<?php
declare(strict_types=1);

namespace Tests\Kata;

use ApprovalTests\Approvals;
use Kata\GildedRose;
use Kata\Item;
use PHPUnit\Framework\TestCase;

class GildedRoseApprovalTest extends TestCase
{
    public function testUpdateQuality(): void
    {
        $items = [
            new Item('normal', 10, 20),
            new Item('normal', 0, 20),
            new Item('Aged Brie', 10, 20),
            new Item('Backstage passes to a TAFKAL80ETC concert', 15, 20),
            new Item('Backstage passes to a TAFKAL80ETC concert', 9, 20),
            new Item('Backstage passes to a TAFKAL80ETC concert', 4, 20),
            new Item('Backstage passes to a TAFKAL80ETC concert', 0, 20),
            new Item('Sulfuras, Hand of Ragnaros', 0, 80),
            new Item('normal', 10, 0),
            new Item('Aged Brie', 10, 50),
            new Item('normal', 10, 20),
            new Item('Sulfuras, Hand of Ragnaros', 0, 80),
            new Item('Aged Brie', 0, 20),
            new Item('normal', 0, 0)
        ];

        $gildedRose = new GildedRose($items);

        for ($i = 0; $i < 10; $i++) {
            $gildedRose->updateQuality();
        }

        Approvals::verifyList($items);
    }
}