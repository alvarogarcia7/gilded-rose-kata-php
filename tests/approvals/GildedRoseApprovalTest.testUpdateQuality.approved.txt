[0] -> normal, 0, 10
[1] -> normal, -10, 0
[2] -> Aged Brie, 0, 30
[3] -> Backstage passes to a TAFKAL80ETC concert, 5, 35
[4] -> Backstage passes to a TAFKAL80ETC concert, -1, 0
[5] -> Backstage passes to a TAFKAL80ETC concert, -6, 0
[6] -> Backstage passes to a TAFKAL80ETC concert, -10, 0
[7] -> Sulfuras, Hand of Ragnaros, 0, 80
[8] -> normal, 0, 0
[9] -> Aged Brie, 0, 50
[10] -> normal, 0, 10
[11] -> Sulfuras, Hand of Ragnaros, 0, 80
[12] -> Aged Brie, -10, 40
[13] -> normal, -10, 0
